src/web/services/search.service.ts

```diff
@@ -294,15 +294,19 @@ export class SearchService {
       registeredAt: '',
       registrationLink: '',
       showLinks: false,
+      hasRegistered: false,
     };
 
     const { registrationKey, createdAt, registeredAt, name, institute, email }: AccountRequest = accountRequest;
 
     const timezone: string = this.timezoneService.guessTimezone() || 'UTC';
     accountRequestResult.createdAt = this.formatTimestampAsString(createdAt, timezone);
-    accountRequestResult.registeredAt = registeredAt
-        ? this.formatTimestampAsString(registeredAt, timezone)
-        : 'Not Registered Yet' ;
+    if (registeredAt) {
```

Comment:
is there a case where `registeredAt` is undefined/null ? I thought it is intialized to `'Not Registered Yet'`?

Reply:
`registeredAt` retrieved from API is of type number and is undefined if the instructor has not registered yet. `registeredAt` in the `AccountRequestSearchResult` is of type string is replaced by the string 'Not Registered Yet' if `registeredAt` taken from API is undefined.

I think there is confusion here because I am using two variables with the same name but of different types, I'll change the name of one to avoid ambiguity.